{Release} Upgrade to Azure CLI 2.39.0 (#23378)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.39.0
++++++

**ACR**

* [BREAKING CHANGE] Update manifest list-referrers to comply with RC1 ORAS spec (#23132)
* `az acr update`: Update networkRuleSet.defaultAction to deny when `--public-network-enabled` is disabled (#23251)
* Fix #23340: `az acr task credential add`: Fix crashes when given a password but no username (#23345)

**AD**

* `az ad app federated-credential`: Federated identity credential GA (#23122)

**Advisor**

> * Fix #11070: Fix `az advisor recommendation disable` NoneType error (#23260)

Reviewer: ```suggestion
* Fix #11070: `az advisor recommendation disable`: Fix NoneType error (#23260)
```

* `az aks create`: Add new parameter `--host-group-id` to support Azure dedicated host (#23324)
* `az aks nodepool add`: Add new parameter `--host-group-id` to support Azure dedicated host (#23324)
* `az aks create/update`: Add new parameters `--enable-azure-keyvault-kms`, `--azure-keyvault-kms-key-id`, `--azure-keyvault-kms-key-vault-network-access`, `--azure-keyvault-kms-key-vault-resource-id` and `--disable-azure-keyvault-kms` to support Key Management Service feature with Azure Key Vault (#23331)
* `az aks create`: Add `--network-plugin=none` support for BYO CNI (#23344)
* `az aks create/update`: Add parameter `--http-proxy-config` to support setting HTTP Proxy configuration (#23352)

**App Service**

* Fix #23135: `az functionapp plan create`: Add validation for the valid value of `--number-of-workers` option (#23153)
* `az functionapp/logicapp create`: Add new `--https-only` parameter (#23213)
* `az functionapp/webapp create`: Allow vnet integration for basic and elastic premium SKUs (#23213)
* `az webapp list-runtimes`: Add Java 17 Support (#23353)
* `az webapp create`: Add Java 17 Support (#23353)
* `az webapp up`: Add Java 17 Support (#23353)

> * `functionapp deployment github-actions add`: Add command to create GitHub actions to deploy to a Function App (#23326)
> * `functionapp deployment github-actions remove`: Add command to remove Function App GitHub actions (#23326)
> * `webapp deployment github-actions`: Add validation to ensure app is Web App (#23326)

Reviewer 2: ```suggestion
* `az functionapp deployment github-actions add`: Add command to create GitHub actions to deploy to a Function App (#23326)
* `az functionapp deployment github-actions remove`: Add command to remove Function App GitHub actions (#23326)
* `az webapp deployment github-actions`: Add validation to ensure app is Web App (#23326)
```

* `az network private-endpoint-connection`: Enable private link support for provider `Microsoft.KubernetesConfiguration/privateLinkScopes` (#23172)
* `az network private-endpoint-connection`: Enable private link support for provider `Microsoft.Dashboard/grafana` (#22298)
* `az network dns zone export`: Add support for ALIAS record (#23209)
* `az network dns zone import`: Add support for ALIAS record (#23274)
* `az network application-gateway waf-policy custom-rule match-condition add`: Add validation for WAF custom rule condition (#23137)
* `az network watcher flow-log`: Add support for `--vnet`, `--subnet`, `--nic` as target ID (#23231)
* `az network private-endpoint create`: Add an example for creating with ASGs (#23100)

**Packaging**

* Drop CentOS 7 RPM package (#23047)
* Drop Python 3.6 support (#23102)
* Build RPM for Fedora (#22945)
* Drop Ubuntu 21.10 Impish Indri DEB package (#23103)

**Profile**

* `az account list`: Add `TenantId` column to table output (#23214)

**RDBMS**

* `az mysql flexible-server server-logs`: Add server logs for MySQL Flexible Server (#23185)

**Service Connector**

> * `az spring connection create eventhubs`: Add new param `--client-type kafka-springBoot` (#23136)

Reviewer: ```suggestion
* `az spring connection create eventhubs`: Add new parameter `--client-type kafka-springBoot` (#23136)
```

* `az webapp connection create`: Add `--config-connstr` to support webapp connection strings (#23288)

> * `Az webapp connection create`: Use webapp name and rg from config (#23313)

Reviewer 2: ```suggestion
* `az webapp connection create`: Use webapp name and rg from config (#23313)
```

> * `Az webapp connection create`: Use webapp name and rg from config (#23313)

Reviewer 3: ```suggestion
* `az webapp connection create`: Use webapp name and rg from config (#23313)
```

> * `Az webapp connection create`: Use webapp name and rg from config (#23313)

Reviewer: ```suggestion
* `Az webapp connection create`: Use webapp name and resource group from config (#23313)
```

> * `az webapp connection create`: Use webapp name and rg from config (#23313)

Reviewer: ```suggestion
* `az webapp connection create`: Use webapp name and resource group from config (#23313)
```

**SQL**

* `az sql log-replay stop`: Drop DB only if it was created with LRS (#22939)

**Storage**

* `az storage fs undelete-path`: Encode `--deleted-path-name` automatically (#23113)
* Fix #23179: `az storage file upload/upload-batch`: Fix `--content-md5` for upload, ignore `--content-md5` for upload-batch (#23207)
* `az storage file show`: Fix JSON error when content-md5 is not None (#23207)
* `az storage blob/file update`: Fix `--content-md5` TypeError (#23253)

> * `az storage container policy create`: No longer uses default value for start and expiry time (#23259)

Reviewer 2: ```suggestion
* `az storage container policy create`: No longer use default value for start and expiry time (#23259)
```
